File: src/java/Reserva.java
package java;

import java.util.Date;

/**
 * @author javier garcia
 * Clase que representa una reserva de un libro en la biblioteca.
 */
public class Reserva {
    private Usuario usuario;
    private Libro libro;
    private Date fechaReserva;
    private Date fechaDevolucion;

    /**
     * Constructor de la clase Reserva.
     * @param usuario El usuario que realiza la reserva.
     * @param libro El libro que se reserva.
     * @param fechaReserva La fecha en que se realiza la reserva.
     * @param fechaDevolucion La fecha estimada de devolución del libro.
     */
    public Reserva(Usuario usuario, Libro libro, Date fechaReserva, Date fechaDevolucion) {
        this.usuario = usuario;
        this.libro = libro;
        this.fechaReserva = fechaReserva;
        this.fechaDevolucion = fechaDevolucion;
    }

    /**
     * Obtiene el usuario que realizó la reserva.
     * @return El usuario de la reserva.
     */
    public Usuario getUsuario() {
        return usuario;
    }

    /**
     * Obtiene el libro que se reservó.
     * @return El libro reservado.
     */
    public Libro getLibro() {
        return libro;
    }

    /**
     * Obtiene la fecha en que se realizó la reserva.
     * @return La fecha de reserva.
     */
    public Date getFechaReserva() {
        return fechaReserva;
    }

    /**
     * Obtiene la fecha estimada de devolución del libro.
     * @return La fecha de devolución estimada.
     */
    public Date getFechaDevolucion() {
        return fechaDevolucion;
    }

    /**
     * Convierte el objeto Reserva en una representación de cadena de texto.
     * @return Una cadena de texto que representa la reserva.
     */
    @Override
    public String toString() {
        return "Reserva{" +
                "usuario=" + usuario.getNombre() +
                ", libro=" + libro.getTitulo() +
                ", fechaReserva=" + fechaReserva +
                ", fechaDevolucion=" + fechaDevolucion +
                '}';
    }
}

File: src/js/Reserva.js
/**
 * @author javier garcia
 * Clase que representa una reserva de un libro en la biblioteca.
 * @class
 */
class Reserva {
  /**
   * Constructor de la clase Reserva.
   * @constructor
   * @param {Usuario} usuario - El usuario que realiza la reserva.
   * @param {Libro} libro - El libro que se reserva.
   * @param {Date} fechaReserva - La fecha en que se realiza la reserva.
   * @param {Date} fechaDevolucion - La fecha estimada de devolución del libro.
   */
  constructor(usuario, libro, fechaReserva, fechaDevolucion) {
    this.usuario = usuario;
    this.libro = libro;
    this.fechaReserva = fechaReserva;
    this.fechaDevolucion = fechaDevolucion;
  }

  /**
   * Obtiene el usuario que realizó la reserva.
   * @returns {Usuario} El usuario de la reserva.
   */
  getUsuario() {
    return this.usuario;
  }

  /**
   * Obtiene el libro que se reservó.
   * @returns {Libro} El libro reservado.
   */
  getLibro() {
    return this.libro;
  }

  /**
   * Obtiene la fecha en que se realizó la reserva.
   * @returns {Date} La fecha de reserva.
   */
  getFechaReserva() {
    return this.fechaReserva;
  }

  /**
   * Obtiene la fecha estimada de devolución del libro.
   * @returns {Date} La fecha de devolución estimada.
   */
  getFechaDevolucion() {
    return this.fechaDevolucion;
  }

  /**
   * Convierte el objeto Reserva en una representación de cadena de texto.
   * @returns {string} Una cadena de texto que representa la reserva.
   */
  toString() {
    return (
      `Reserva { Usuario: ${this.usuario.getNombre()}, Libro: ${this.libro.getTitulo()}, ` +
      `Fecha de Reserva: ${this.fechaReserva}, Fecha de Devolución: ${this.fechaDevolucion} }`
    );
  }
}

// Exporta la clase para que pueda ser utilizada en otros módulos si es necesario.
export default Reserva;
